go/cmd/multigres/command/cluster/up.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2025 The Multigres Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import (
+	"fmt"
+	"net"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/multigres/multigres/go/servenv"
+
+	"github.com/spf13/cobra"
+)
+
+// checkEtcdConnectivity checks if etcd is reachable at the given address
+func checkEtcdConnectivity(address string) error {
+	conn, err := net.DialTimeout("tcp", address, 5*time.Second)
+	if err != nil {
+		return fmt.Errorf("failed to connect to etcd at %s: %w", address, err)
+	}
+	defer conn.Close()
+	return nil
+}
+
+// startEtcdContainer starts an etcd container using Docker
+func startEtcdContainer(address string) error {
+	fmt.Printf("Starting etcd container for address: %s\n", address)
+
+	// Extract port from address (assuming format like "localhost:2379")
+	parts := strings.Split(address, ":")
+	if len(parts) != 2 {
+		return fmt.Errorf("invalid address format: %s (expected host:port)", address)
+	}
+	port := parts[1]
+
+	// Check if docker is available
+	if err := CheckDockerAvailable(); err != nil {
+		return fmt.Errorf("docker not available: %w", err)
+	}
+
+	// Create Multigres network for better container orchestration
+	if err := CreateMultigresNetwork(); err != nil {
+		return fmt.Errorf("failed to create network: %w", err)
+	}
+
+	containerName := EtcdContainerName
+
+	// Check if container already exists
+	if IsContainerRunning(containerName) {
+		fmt.Printf("etcd container '%s' is already running\n", containerName)
+		return nil
+	}
+
+	// Remove any stopped container with the same name
+	if err := RemoveContainer(containerName); err != nil {
+		return fmt.Errorf("failed to remove container: %w", err)
+	}
+
+	// Calculate peer port (client port + 1 for convention)
+	clientPort := port
+	peerPort := fmt.Sprintf("%d", 2380) // Keep peer port standard for now
+
+	// Build Docker command with proper grouping labels
+	args := []string{
+		"run", "-d",
```

Comment:
I did not validate if these flags are exactly what we want, but it works in terms of bringing etcd up locally.